### PR TITLE
fix: remove subscription consumer instead of re-registering

### LINKED ISSUE #9123: Removing `Subscription` consumer re-registers the consumer instead of removing
**Describe the bug**

https://github.com/camunda/zeebe/blob/1edfe4f762ef700885859bf87e1370963d3448a1/dispatcher/src/main/java/io/camunda/zeebe/dispatcher/Subscription.java#L83-L86

this should obviously be a call to `removeConsumer`, not `registerConsumer`. As far as I can tell this bug has existed since at least 2018. 

Removing a subscription consumer is triggered when an ActorTask is closed (successfully or due to failure).

### PR DESCRIPTION
**Pull request by lenaschoenburg**
This was probably not an issue because it was mitigated by other circumstances.

closes #9123

### CODE REVIEW

**Review by ChrisKujawa — COMMENTED** [review]
Thanks @oleschoenburg could we add a test to verify that? 

I saw there is a method to get the conditions so we could simply check that it is empty after removing. WDYT?

**Comment by Zelldon** [comment]
BTW congrats to your first Dispatcher bug fix :laughing:

**Comment by oleschoenburg** [comment]
I've added a test that verifies that subscription consumers are not signaled after removing them from the subscription. I think this is a good test case to add since it verifies the behavior (no more signals) and not the implementation (the consumer array is empty).

File: dispatcher/src/test/java/io/camunda/zeebe/dispatcher/SubscriptionConsumerTest.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  void consumersAreNotSignaledAfterRemoving() {
+    // given
+    final var consumer = mock(ActorCondition.class);
+    final var subscription =
+        new Subscription(
+            mock(AtomicPosition.class),
+            mock(AtomicPosition.class),
+            0,
+            "",
+            consumer,
```
Comment (Zelldon) [inline dispatcher/src/test/java/io/camunda/zeebe/dispatcher/SubscriptionConsumerTest.java:28]: :x: This doesn't do what you expect.

**Review by ChrisKujawa — CHANGES_REQUESTED** [review]
Thanks Ole :+1: was also thinking about that but not sure how easy it is. :) Good that you wrote one :+1: 

Small thing we have to fix on the test :)

File: dispatcher/src/test/java/io/camunda/zeebe/dispatcher/SubscriptionConsumerTest.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            mock(LogBuffer.class));
+
+    // when
+    subscription.removeConsumer(consumer);
```
[the remark this answers is not in the document]
Comment (oleschoenburg) [reply dispatcher/src/test/java/io/camunda/zeebe/dispatcher/SubscriptionConsumerTest.java:32]: Oh, I see. I only reverted the fix and verified that the test fails. I'll fix this :+1:

**Review by ChrisKujawa — APPROVED** [review]
Thanks :+1:

**Pull request merged**